src/main/java/com/yahoo/bullet/query/expressions/BinaryExpression.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright 2019, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.query.expressions;
+
+import com.yahoo.bullet.common.BulletException;
+import com.yahoo.bullet.querying.evaluators.BinaryEvaluator;
+import com.yahoo.bullet.querying.evaluators.Evaluator;
+import lombok.Getter;
+
+import java.util.Objects;
+
+import static com.yahoo.bullet.query.expressions.Operation.BINARY_OPERATIONS;
+
+/**
+ * An expression that takes two operands and a binary operation. These fields are required; however, an optional
+ * primitive type may be specified.
+ *
+ * Infix and prefix binary operations are differentiated in the naming scheme.
+ */
+@Getter
+public class BinaryExpression extends Expression {
+    private static final long serialVersionUID = -7911485746578844403L;
+
+    public enum Modifier {
```

Comment:
I made a note of this in the BinaryEvaluator but I don't think this should be needed. I can't see a point for it in bullet-core anyway.